dali/pipeline/operators/crop/bbox_crop.h

```diff
@@ -0,0 +1,251 @@
+
+      return intersection_area /
+             static_cast<float>(area_ + other.area_ - intersection_area);
+    }
+
+    const float left_, top_, right_, bottom_, area_;
+  };
+
+  using Crop = Rectangle;
+  using BoundingBox = Rectangle;
+  using BoundingBoxes = std::vector<Rectangle>;
+
+ public:
+  explicit inline BBoxCrop(const OpSpec &spec)
+      : Operator<CPUBackend>(spec),
+        thresholds_{[&spec]() {
+          return spec.GetRepeatedArgument<float>("thresholds");
+        }()},
+        scaling_bounds_{[&spec]() {
+          return Bounds(spec.GetRepeatedArgument<float>("scaling"));
+        }()},
+        aspect_ratio_bounds_{[&spec]() {
+          return Bounds(spec.GetRepeatedArgument<float>("aspect_ratio"));
+        }()}
+
+  {
+    DALI_ENFORCE(!thresholds_.empty(),
+                 "At least one threshold value must be provided");
+
+    for (const auto &threshold : thresholds_) {
+      DALI_ENFORCE(0.0 <= threshold, "Threshold value must be >= 0.0");
+      DALI_ENFORCE(threshold <= 1.0, "Threshold value must be <= 1.0");
+    }
+  }
+
+  virtual ~BBoxCrop() = default;
+
+ protected:
+  void RunImpl(SampleWorkspace *ws, const int idx) {
+    const auto &image = ws->Input<CPUBackend>(0);
+    const auto &bounding_boxes = ws->Input<CPUBackend>(1);
+
+    const auto minimum_overlap = SelectMinimumOverlap();
+
+    const auto prospective_crop =
+        FindProspectiveCrop(image, bounding_boxes, minimum_overlap);
+
+    WriteCropToOutput(ws, prospective_crop.first);
+    WriteBoxesToOutput(ws, prospective_crop.second);
+  }
+
+  void WriteCropToOutput(SampleWorkspace *ws, const Crop &crop) {
+    // Copy the anchor to output 0
+    auto *anchor_out = ws->Output<CPUBackend>(0);
+    anchor_out->Resize({2});
+
+    auto *anchor_out_data = anchor_out->mutable_data<float>();
+    anchor_out_data[0] = crop.left_;
+    anchor_out_data[1] = crop.top_;
+
+    // Copy the offsets to output 1
+    auto *offsets_out = ws->Output<CPUBackend>(1);
+    offsets_out->Resize({2});
+
+    auto *offsets_out_data = offsets_out->mutable_data<float>();
+    offsets_out_data[0] = crop.right_ - crop.left_;
+    offsets_out_data[1] = crop.bottom_ - crop.top_;
+  }
+
+  void WriteBoxesToOutput(SampleWorkspace *ws,
+                          const BoundingBoxes &bounding_boxes) {
+    auto *bbox_out = ws->Output<CPUBackend>(1);
+    bbox_out->Resize({static_cast<int>(bounding_boxes.size()), kBboxSize});
+
+    auto *bbox_out_data = bbox_out->mutable_data<float>();
+    for (size_t i = 0; i < bounding_boxes.size(); ++i) {
+      auto *output = bbox_out_data + i * kBboxSize;
+      output[0] = bounding_boxes[i].left_;
+      output[1] = bounding_boxes[i].top_;
+      output[2] = bounding_boxes[i].right_;
+      output[3] = bounding_boxes[i].bottom_;
+    }
+  }
+
+  float SelectMinimumOverlap() {
+    static std::uniform_int_distribution<> sampler(0, thresholds_.size());
+    return thresholds_[sampler(rd_)];
```

Comment:
Why we choose that randomly - I think there is no mention about that in API.
Shoudln't it be or one value for all images or list/tensor of values per image each?

Reply:
Each image should have applied a threshold from those that are supplied. Added this info to API